File: server/src/main/java/com/xcore/application/authentication/controllers/api/CurrentAuthInfoResponse.java
package com.xcore.application.authentication.controllers.api;

import com.xcore.server.controllers.rest.general.api.Response;
import lombok.Data;
import org.springframework.security.core.Authentication;

@Data
public class CurrentAuthInfoResponse extends Response {

    private Boolean authenticated = null;


    public CurrentAuthInfoResponse(Authentication authentication) {
        this.authenticated = authentication.isAuthenticated();
    }

}

File: server/src/main/scala/com/xcore/application/authentication/controllers/api/CurrentAuthInfoResponse.scala
package com.xcore.application.authentication.controllers.api;

import com.xcore.server.controllers.rest.general.api.Response;
import org.springframework.security.core.Authentication;

class CurrentAuthInfoResponse (authentication: Authentication) extends Response {

  private var authenticated: Boolean = false;

  def CurrentAuthInfoResponse(authentication: Authentication) {
    this.authenticated = authentication.isAuthenticated;
  }

  def isAuthenticated() = this.authenticated;

}

